notebook/autobuild_function_calling.ipynb

```diff
@@ -0,0 +1,522 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# AutoBuild Agents function calling \n",
+    "By: [Krishna Shedbalkar](https://github.com/krishnashed/)\n",
+    "\n",
+    "In this notebook, we introduce a way for Agents created using `Autobuild` to do function calling. Developers can specify a function, function name and function description which will thereafter be assigned and executed by the most suitable agent created using AutoBuild."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Requirement\n",
+    "\n",
+    "AutoBuild require `pyautogen[autobuild]`, which can be installed by the following command:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install pyautogen[autobuild]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 1: Prepare configuration and some useful functions\n",
+    "\n",
+    "Prepare a `config_file_or_env` for assistant agent to limit the choice of LLM you want to use in this task. This config can be a path of json file or a name of environment variable. A `default_llm_config` is also required for initialize the specific config of LLMs like seed, temperature, etc..."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import autogen\n",
+    "from autogen.agentchat.contrib.agent_builder import AgentBuilder\n",
+    "\n",
+    "config_file_or_env = \"OAI_CONFIG_LIST\"\n",
+    "config_list = autogen.config_list_from_json(config_file_or_env, filter_dict={\"model\": [\"gpt-4-1106-preview\", \"gpt-4\"]})\n",
+    "llm_config = {\n",
+    "    \"config_list\": config_list,\n",
+    "    \"timeout\": 120,\n",
+    "}\n",
+    "\n",
+    "\n",
+    "def start_task(execution_task: str, agent_list: list):\n",
+    "    group_chat = autogen.GroupChat(agents=agent_list, messages=[], max_round=12)\n",
```

Comment:
You can prevent User_console from being called multiple times by adding `allow_repeat_speaker=[list of agent except user_proxy]` when initializing group chat.